csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/AppInsights/MyAppInsightsLoggerMiddleware.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace NLP_With_Dispatch_Bot.AppInsights
+{
+    /// <summary>
+    /// Middleware for logging incoming, outgoing, updated or deleted Activity messages into Application Insights.
```

Comment:
>Activity [](start = 70, length = 8)

see tag

Reply:
Removing app insights, this is going away